### ISSUE TITLE
Model change detected because of newlines across Windows/Unix

### ISSUE THREAD

**Issue by roji** [dotnet/efcore#35382]
SQL strings in various parts of the model (e.g. default value SQL, filters...) may embed newlines, and if they do, falsely trigger a model change (and therefore a PendingModelChangesWarning). We should consider normalizing newlines for the purpose of model diffing to prevent this.

Reported by @dcby in https://github.com/dotnet/efcore/issues/35285#issuecomment-2559892621

**Comment by nielsNocore** [comment]
We had the exacte same issue, in a multi line constraint. 
was really hard to find what this was. as our dev machines are all windows. the build machine is Linux. and the database updater job was failing, with the error message not good enough to know what was going on. 
'Microsoft.EntityFrameworkCore.Migrations.PendingModelChangesWarning' exception.

`An error was generated for warning 'Microsoft.EntityFrameworkCore.Migrations.PendingModelChangesWarning': The model for context 'DbContext' has pending changes. Add a new migration before updating the database. This exception can be suppressed or logged by passing event ID 'RelationalEventId.PendingModelChangesWarning' to the 'ConfigureWarnings' method in 'DbContext.OnConfiguring' or 'AddDbContext'.`

Took us quite some time to find it was related to a multiline constraint

Hoping this will be fixed any time soon.

**Comment by Shane32** [comment]
Agree; this is very frustrating.  It cannot be caught in development since the code was written there and so the line endings match.  It doesn't get caught in our linux CI tests because the database doesn't exist there; the database is created for tests not migrated.  It wouldn't get caught in Windows CI tests because the line endings would match.  So it only gets caught when pushing to production.

**Comment by w7rus** [comment]
Got the same, when were pushing to linux dev server. Migration was made on windows, and there was pending changes also on mac.

Just transformed all content into a regular string from multi-line one.
